invenio/modules/search/static/js/search/facets_menu.js

```diff
@@ -0,0 +1,95 @@
+/*
+ * This file is part of Invenio.
+ * Copyright (C) 2015 CERN.
+ *
+ * Invenio is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * Invenio is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Invenio; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+
+/* A file containing the FlightJS component handling the faceting panel. */
+define([
+        'jquery',
+        'flight/lib/component',
+],
+function(
+    $,
+    defineComponent) {
+  "use strict";
+  // return flightJS component
+  return defineComponent(FacetsMenu);
+
+  /**
+   * FacetsMenu is a FlightJS component for handling facet menu.
+   *
+   * NOTE: it has to be attached to the DOM before the Search component so that
+   * it can listen to facetsSet event.
+   */
+  function FacetsMenu() {
+    this.attributes({
+      includeFacetSelector: 'input.include-facet'
+    });
+
+    /**
+     * Callback called when a facet's value has been included in query filters.
+     * It will broadcast the corresponding event so that other components may
+     * update.
+     *
+     * @param {} ev 
+     * @param {} data
+     */
+    function facetChange(ev, data) {
+      this.trigger(document, 'facetChange', {
+        name: data.el.name,
+        value: data.el.value,
+        state: data.el.checked ? 'include' : 'default',
+      });
+    }
+
+    /**
+     * Initialize the component
+     */
+    this.init = function() {
+      var cpn = this;
```

Comment:
Just a minor thing. In recent JS files added to Invenio the pattern for this has usually been `var that = this` (ala Crockford/jQuery). I do not mind either way, but it would be good if this is standard across all new JS we add. See also discussion in #1769